nf_core/modules/lint/module_version.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+"""
+Verify that a module has a correct entry in the modules.json file
+"""
+
+import logging
+import os
+import json
+import re
+import questionary
+import nf_core
+import sys
+
+import nf_core.modules.module_utils
+
+log = logging.getLogger(__name__)
+
+
+def module_version(module_lint_object, module):
+    """
+    Verify that the module has a version (git_sha) specified in the
+    modules.json file and checks whether a new version is available
+    """
+    modules_json_path = os.path.join(module_lint_object.dir, "modules.json")
+
+    # Verify that a git_sha exists in the `modules.json` file for this module
+    try:
+        git_sha = module_lint_object.modules_json["modules"][module.module_name]["git_sha"]
+        module.git_sha = git_sha
+        module.passed.append(("git_sha", "Found git_sha entry in `modules.json`", modules_json_path))
+    except KeyError:
```

Comment:
Both cases would be captured by the `except` statement though, so it would be a matter of having an additional message like "missing module entry" and "missing git_sha entry".
Personally I think one message is enough, because a missing `git_sha` basically suggets that there is no module entry (as that always ships with a `git_sha` with the tools that you developed I think).
But